src/tests/SimTest.cpp

```diff
@@ -152,33 +198,56 @@ void SimTest::reconfigure() {
   cfg2.activeSceneName = skokloster;
   simulator.reconfigure(cfg2);
   CORRADE_VERIFY(pathfinder != simulator.getPathFinder());
+
+  // test using MM ctor
+  SimulatorConfiguration cfg_mm;
+  cfg_mm.activeSceneName = vangogh;
+  MetadataMediator::ptr MM = MetadataMediator::create(cfg_mm);
+  Simulator simulator_mm(MM);
+  PathFinder::ptr pathfinder_mm = simulator_mm.getPathFinder();
+  simulator_mm.reconfigure(cfg_mm);
+  CORRADE_VERIFY(pathfinder_mm == simulator_mm.getPathFinder());
+  SimulatorConfiguration cfg2_mm;
+  cfg2_mm.activeSceneName = skokloster;
+  simulator_mm.reconfigure(cfg2_mm);
+  CORRADE_VERIFY(pathfinder_mm != simulator_mm.getPathFinder());
 }
 
 void SimTest::reset() {
+  auto testReset = [&](Simulator& simulator) {
```

Comment:
Because all test macros (which take care of extracting the function name) are inside the lambda, the function won't be marked correctly in the output. I'd do 

```cpp
void SimTest::reset() {
  CORRADE_VERIFY(true);

  auto testReset = [&](Simulator& simulator) {
```

to work around that :)